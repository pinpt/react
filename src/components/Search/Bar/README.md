### A component for the search bar in the header

[Full Docs](https://react.preview.pinpoint.com/?path=/docs/components-search-search-bar) | [Github](https://github.com/pinpt/react/tree/master/src/components/Search/Bar)

The search component can use either a change handler or a submit handler (or both) depending on the desired behavior.

#### Theming

The parent className for styling this component is `.Pinpoint.Search.Bar`

<details>
	<summary>Base Theme Styles</summary>

```css
.Pinpoint.Search.Bar {
	@apply border rounded-full flex items-center w-48 h-8 overflow-hidden transition-colors duration-200;
	border-color: var(--header-control-color);
	color: var(--header-control-color);
}

.Pinpoint.Search.Bar .icon {
	@apply ml-3 mr-2;
}

.Pinpoint.Search.Bar .input {
	@apply w-full border-0 py-1 pr-2 outline-none bg-transparent;
	color: var(--page-text-color);
}

.Pinpoint.Search.Bar .input::placeholder {
	@apply opacity-30 transition-colors duration-200;
	color: var(--page-text-color);
}
```

</details>
