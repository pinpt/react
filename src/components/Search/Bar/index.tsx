import { ChangeEvent, FormEvent, useCallback, useState } from 'react';

export interface ISearchBarProps {
	className?: string;
	defaultValue?: string;
	onChange?: (value: string) => void;
	onSubmit?: (value: string) => void;
}

const Bar = (props: ISearchBarProps) => {
	const { className = '', onChange, onSubmit, defaultValue = '' } = props;
	const [value, setValue] = useState(() => defaultValue);

	const handleInputChange = useCallback(
		(e: ChangeEvent<HTMLInputElement>) => {
			const val = e.target.value;
			setValue(val);
			onChange?.(val);
		},
		[onChange]
	);

	const handleFormSubmit = useCallback(
		(e: FormEvent<HTMLFormElement>) => {
			e.preventDefault();
			e.stopPropagation();
			onSubmit?.(value);
			return false;
		},
		[value, onSubmit]
	);

	return (
		<div className={`Pinpoint Search Bar ${className}`}>
			<svg
				className="icon"
				width={16}
				height={16}
				aria-hidden="true"
				focusable="false"
				data-prefix="fas"
				data-icon="search"
				role="img"
				xmlns="http://www.w3.org/2000/svg"
				viewBox="0 0 512 512"
			>
				<path
					fill="currentColor"
					d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
				></path>
			</svg>
			<form className="form" onSubmit={handleFormSubmit}>
				<input className="input" placeholder="Search" value={value} onChange={handleInputChange} />
			</form>
		</div>
	);
};

export default Bar;
