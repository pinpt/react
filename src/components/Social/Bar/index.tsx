import React from 'react';
export interface ISocialBarProps {
	className?: string;
	children: any;
}

const Bar = (props: ISocialBarProps) => {
	const { className = '', children } = props;
	return <div className={`Pinpoint Social Bar ${className}`}>{children}</div>;
};

export default Bar;
