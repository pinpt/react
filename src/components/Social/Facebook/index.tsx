import ActionLink from '../../Internal/ActionLink';
import { ISocialProps } from '../types';

const Facebook = (props: ISocialProps) => {
	const { className = '', sharing, ...rest } = props;

	return (
		<ActionLink className={`Social Item Facebook ${sharing ? 'sharing' : ''} ${className}`} {...rest}>
			{sharing ? (
				<svg
					width={10}
					height={16}
					aria-hidden="true"
					focusable="false"
					data-prefix="fab"
					data-icon="facebook-f"
					role="img"
					xmlns="http://www.w3.org/2000/svg"
					viewBox="0 0 320 512"
				>
					<path
						fill="currentColor"
						d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
					></path>
				</svg>
			) : (
				<svg
					width={14}
					height={14}
					aria-hidden="true"
					focusable="false"
					data-prefix="fab"
					data-icon="facebook"
					role="img"
					xmlns="http://www.w3.org/2000/svg"
					viewBox="0 0 512 512"
				>
					<path
						fill="currentColor"
						d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.78 90.69 226.38 209.25 245V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.31 482.38 504 379.78 504 256z"
					></path>
				</svg>
			)}
		</ActionLink>
	);
};

export default Facebook;
