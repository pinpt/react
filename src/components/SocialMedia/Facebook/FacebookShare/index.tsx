import React from 'react';
import useLinkTracking from '../../../../lib/hooks/useLinkTracking';
import ActionLink from '../../../Internal/ActionLink';
import { ISocialMediaShareProps } from '../../types';

const FacebookShare = (props: ISocialMediaShareProps) => {
	const { className = '', href: _href, ...rest } = props;

	const href = useLinkTracking(_href, 'facebook');

	return (
		<ActionLink
			className={`Pinpoint SocialMedia ShareItem FacebookShare ${className}`}
			title="Twitter"
			href={`https://facebook.com/sharer/sharer.php?u=${encodeURIComponent(href)}`}
			{...rest}
		>
			<svg
				width={10}
				height={16}
				aria-hidden="true"
				focusable="false"
				data-prefix="fab"
				data-icon="facebook-f"
				role="img"
				xmlns="http://www.w3.org/2000/svg"
				viewBox="0 0 320 512"
			>
				<path
					fill="currentColor"
					d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
				></path>
			</svg>
		</ActionLink>
	);
};

export default FacebookShare;
