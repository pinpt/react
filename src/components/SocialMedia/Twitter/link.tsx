import React from 'react';
import ActionLink from '../../Internal/ActionLink';
import { ISocialMediaLinkProps } from '../types';
import Icon from './icon';

const TwitterLink = (props: ISocialMediaLinkProps) => {
	const { className = '', ...rest } = props;

	return (
		<ActionLink className={`Pinpoint SocialMedia Item Twitter link ${className}`} title="Twitter" {...rest}>
			<Icon />
		</ActionLink>
	);
};

export default TwitterLink;
