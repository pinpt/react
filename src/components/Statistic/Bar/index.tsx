import Claps from '../Claps';
import Views from '../Views';

export interface IStatisticsBarProps {
	className?: string;
	claps?: number;
	views?: number;
}

const Bar = (props: IStatisticsBarProps) => {
	const { className, claps, views } = props;
	return (
		<div className={`Pinpoint Statistic Bar wrapper ${className ?? ''}`}>
			<Views count={views ?? 0} />
			<Claps count={claps ?? 0} />
		</div>
	);
};

export default Bar;
