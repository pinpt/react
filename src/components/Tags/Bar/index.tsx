import { cancelEvent } from '../../../lib';
import Item from '../Item';
import React from 'react';

export interface ITagBarProps {
	className?: string;
	removable?: boolean;
	tags: string[];
	onClick?: (tag: string) => void;
}

const Bar = (props: ITagBarProps) => {
	const { className = '', tags, onClick, removable } = props;

	return (
		<div className={`Pinpoint Tag Bar ${className}`}>
			{tags.map((tag) => {
				return (
					<Item
						removable={removable}
						key={tag}
						tag={tag}
						onClick={onClick ? (e) => cancelEvent(e, () => onClick(tag)) : undefined}
					/>
				);
			})}
		</div>
	);
};

export default Bar;
