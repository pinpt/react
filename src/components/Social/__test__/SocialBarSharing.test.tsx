import renderer from 'react-test-renderer';
import Bar from '../Bar';
import Email from '../Email';
import Facebook from '../Facebook';
import Github from '../Github';
import Instagram from '../Instagram';
import LinkedIn from '../LinkedIn';
import Twitter from '../Twitter';

test('Test all', () => {
	const component = renderer.create(
		<Bar>
			<Facebook sharing href="https://www.facebook.com/Pinpoint.Engineering" newTab />
			<Twitter sharing href="https://twitter.com/pinpoint_sw" newTab />
			<LinkedIn sharing href="https://linkedin.com/company/pinpoint-software" newTab />
			<Email sharing href="mailto:hello@pinpoint.com" />
		</Bar>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test short', () => {
	const component = renderer.create(
		<Bar>
			<Twitter sharing href="https://twitter.com/pinpoint_sw" newTab />
			<LinkedIn sharing href="https://linkedin.com/company/pinpoint-software" newTab />
		</Bar>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Test single', () => {
	const component = renderer.create(
		<Bar>
			<Twitter sharing href="https://twitter.com/pinpoint_sw" newTab />
		</Bar>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('Custom className', () => {
	const component = renderer.create(
		<Bar className="custom-test">
			<Twitter className="custom-test" sharing href="https://twitter.com/pinpoint_sw" newTab />
		</Bar>
	);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
