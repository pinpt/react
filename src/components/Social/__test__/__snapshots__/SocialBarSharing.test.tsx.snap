// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom className 1`] = `
<div
  className="Pinpoint Social Bar custom-test"
>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Twitter sharing custom-test"
    href="https://twitter.com/pinpoint_sw"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="twitter"
      data-prefix="fab"
      focusable="false"
      height={14}
      role="img"
      viewBox="0 0 512 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Test all 1`] = `
<div
  className="Pinpoint Social Bar "
>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Facebook sharing "
    href="https://www.facebook.com/Pinpoint.Engineering"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="facebook-f"
      data-prefix="fab"
      focusable="false"
      height={16}
      role="img"
      viewBox="0 0 320 512"
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M279.14 288l14.22-92.66h-88.91v-60.13c0-25.35 12.42-50.06 52.24-50.06h40.42V6.26S260.43 0 225.36 0c-73.22 0-121.08 44.38-121.08 124.72v70.62H22.89V288h81.39v224h100.17V288z"
        fill="currentColor"
      />
    </svg>
  </a>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Twitter sharing "
    href="https://twitter.com/pinpoint_sw"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="twitter"
      data-prefix="fab"
      focusable="false"
      height={14}
      role="img"
      viewBox="0 0 512 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        fill="currentColor"
      />
    </svg>
  </a>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item LinkedIn sharing "
    href="https://linkedin.com/company/pinpoint-software"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="linkedin-in"
      data-prefix="fab"
      focusable="false"
      height={16}
      role="img"
      viewBox="0 0 448 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
        fill="currentColor"
      />
    </svg>
  </a>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Email sharing "
    href="mailto:hello@pinpoint.com"
  >
    <svg
      aria-hidden="true"
      data-icon="envelope"
      data-prefix="fas"
      focusable="false"
      height={16}
      role="img"
      viewBox="0 0 512 512"
      width={16}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M502.3 190.8c3.9-3.1 9.7-.2 9.7 4.7V400c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V195.6c0-5 5.7-7.8 9.7-4.7 22.4 17.4 52.1 39.5 154.1 113.6 21.1 15.4 56.7 47.8 92.2 47.6 35.7.3 72-32.8 92.3-47.6 102-74.1 131.6-96.3 154-113.7zM256 320c23.2.4 56.6-29.2 73.4-41.4 132.7-96.3 142.8-104.7 173.4-128.7 5.8-4.5 9.2-11.5 9.2-18.9v-19c0-26.5-21.5-48-48-48H48C21.5 64 0 85.5 0 112v19c0 7.4 3.4 14.3 9.2 18.9 30.6 23.9 40.7 32.4 173.4 128.7 16.8 12.2 50.2 41.8 73.4 41.4z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Test short 1`] = `
<div
  className="Pinpoint Social Bar "
>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Twitter sharing "
    href="https://twitter.com/pinpoint_sw"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="twitter"
      data-prefix="fab"
      focusable="false"
      height={14}
      role="img"
      viewBox="0 0 512 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        fill="currentColor"
      />
    </svg>
  </a>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item LinkedIn sharing "
    href="https://linkedin.com/company/pinpoint-software"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="linkedin-in"
      data-prefix="fab"
      focusable="false"
      height={16}
      role="img"
      viewBox="0 0 448 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M100.28 448H7.4V148.9h92.88zM53.79 108.1C24.09 108.1 0 83.5 0 53.8a53.79 53.79 0 0 1 107.58 0c0 29.7-24.1 54.3-53.79 54.3zM447.9 448h-92.68V302.4c0-34.7-.7-79.2-48.29-79.2-48.29 0-55.69 37.7-55.69 76.7V448h-92.78V148.9h89.08v40.8h1.3c12.4-23.5 42.69-48.3 87.88-48.3 94 0 111.28 61.9 111.28 142.3V448z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;

exports[`Test single 1`] = `
<div
  className="Pinpoint Social Bar "
>
  <a
    className="Pinpoint Internal Link Pinpoint Social Item Twitter sharing "
    href="https://twitter.com/pinpoint_sw"
    rel="noopener noreferrer"
    target="_blank"
  >
    <svg
      aria-hidden="true"
      data-icon="twitter"
      data-prefix="fab"
      focusable="false"
      height={14}
      role="img"
      viewBox="0 0 512 512"
      width={14}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
        fill="currentColor"
      />
    </svg>
  </a>
</div>
`;
